pygen/pygen_src/riscv_asm_program_gen.py

```diff
@@ -437,15 +437,35 @@ def init_floating_point_gpr_with_dpf(self, int_floating_gpr):
         self.instr_stream.extend((li_instr0, slli_instr, li_instr1, or_instr, fmv_instr))
 
     # Get a random single precision floating value
+    @staticmethod
+    def get_randselect0(addr_range1, addr_range2):
+        value = vsc.rand_bit_t(32)
+        with vsc.randomize_with(value): value in vsc.rangelist(addr_range1, addr_range2)
+
     def get_rand_spf_value(self):
-        # TODO randcase
-        value = random.randrange(0, 2**32 - 1)
+        vsc.randselect([
+            (1, lambda: get_randselect0(hex(0x7f80_0000), hex(0xff80_0000))),
```

Comment:
Maybe adding a comment here would be great (about the addr_range1 and addr_range2)